apps/design/backend/src/worker/tasks.ts

```diff
@@ -0,0 +1,68 @@
+import fs from 'fs/promises';
+import JsZip from 'jszip';
+import path from 'path';
+import { z } from 'zod';
+import { throwIllegalValue } from '@votingworks/basics';
+import { layOutAllBallotStyles } from '@votingworks/hmpb-layout';
+import {
+  BallotType,
+  ElectionPackageFileName,
+  getDisplayElectionHash,
+  Id,
+  safeParseJson,
+} from '@votingworks/types';
+
+import { PORT } from '../globals';
+import { BackgroundTask } from '../store';
+import { Workspace } from '../workspace';
+
+async function generateElectionPackage(
+  { assetDirectoryPath, store }: Workspace,
+  { electionId }: { electionId: Id }
+): Promise<void> {
+  const { election, layoutOptions, systemSettings } =
+    store.getElection(electionId);
+  const { electionDefinition } = layOutAllBallotStyles({
+    election,
+    // Ballot type and ballot mode shouldn't change the election definition, so it doesn't matter
+    // what we pass here
+    ballotType: BallotType.Precinct,
+    ballotMode: 'test',
+    layoutOptions,
+  }).unsafeUnwrap();
+  const zip = new JsZip();
+  zip.file(ElectionPackageFileName.ELECTION, electionDefinition.electionData);
+  zip.file(
+    ElectionPackageFileName.SYSTEM_SETTINGS,
+    JSON.stringify(systemSettings, null, 2)
+  );
+
+  const displayElectionHash = getDisplayElectionHash(electionDefinition);
+  const fileName = `election-package-${displayElectionHash}.zip`;
+  const filePath = path.join(assetDirectoryPath, fileName);
+  await fs.writeFile(filePath, await zip.generateAsync({ type: 'nodebuffer' }));
+  store.setElectionPackageUrl({
+    electionId,
+    electionPackageUrl: `http://localhost:${PORT}/${fileName}`,
+  });
+}
+
+export async function processBackgroundTask(
+  workspace: Workspace,
+  { taskName, payload }: BackgroundTask
+): Promise<void> {
+  switch (taskName) {
+    case 'generate_election_package': {
+      const parsedPayload = safeParseJson(
+        payload,
+        z.object({ electionId: z.string() })
+      ).unsafeUnwrap();
+      await generateElectionPackage(workspace, parsedPayload);
```

Comment:
I wonder if we want some kind of transaction management for background tasks? In Arlo, the entire task is one transaction, so if any error occurs it can be rolled back. It doesn't feel like a blocking concern here since there's only one write and it's the last thing in the task, but thought I'd note it for the future.

Reply:
Good nudge! I'm going back and forth right now on whether to build an automatic mechanism vs. giving task handlers flexibility.

Looking ahead to the addition of language and audio files to election packages, I see some value in not automatically wrapping everything in a transaction. If, for example, one speech synthesis request fails because of a Google Cloud blip, we wouldn't want to roll back speech synthesis results already added to the cache in the DB.

I'll think some more about this as I add language and audio files!